src/map/packet_system.cpp

```diff
@@ -1165,7 +1165,9 @@ void SmallPacket0x032(map_session_data_t* const PSession, CCharEntity* const PCh
     if ((PTarget != nullptr) && (PTarget->id == charid))
     {
         ShowDebug(CL_CYAN "%s initiated trade request with %s\n" CL_RESET, PChar->GetName(), PTarget->GetName());
-        if (jailutils::InPrison(PChar) || jailutils::InPrison(PTarget))
+
+        // If player is invisible or if either player is in prison don't allow the trade.
+        if (PChar->StatusEffectContainer->HasStatusEffect(EFFECT_INVISIBLE)) || jailutils::InPrison(PChar) || jailutils::InPrison(PTarget))
```

Comment:
I think you have an extra ")"
if (
    PChar->StatusEffectContainer->HasStatusEffect(EFFECT_INVISIBLE)) <--Right here
 
||
    jailutils::InPrison(PChar)
||
    jailutils::InPrison(PTarget)
)